src/Build/Construction/ProjectElementContainer.cs

```diff
@@ -448,6 +448,23 @@ private void SetElementAsAttributeValue(ProjectElement child)
             ProjectXmlUtilities.SetOrRemoveAttribute(XmlElement, child.XmlElement.Name, value);
         }
 
+        /// <summary>
+        /// If child "element" is actually represented as an attribute, update the name in the corresponding Xml attribute
+        /// </summary>
+        /// <param name="child">A child element which might be represented as an attribute</param>
+        /// <param name="oldName">The old name for the child element</param>
+        internal void UpdateElementName(ProjectElement child, string oldName)
+        {
+            ErrorUtilities.VerifyThrow(Link == null, "External project");
```

Comment:
Please add an error message that would be understandable to anyone reviewing log of crashing msbuild.
(This is inspired by still painful memory of troubleshooting sudden msbuild crashes with error message being `"must be valid"`)

Reply:
I'm not totally sure I get what "Link" even is tbh. This assert is copied from `SetElementAsAttributeValue`.

Reply:
https://github.com/dotnet/msbuild/issues/10695